File: Python/7py 1/dgmf.py
"""
|--------------------------------------------------------------------------
| Don't give me five!
|--------------------------------------------------------------------------
|
"""

def dont_give_me_five(start,end):
   res = 0
   for i in range(start, end+1):
      arr = list(str(i))
      if '5' not in arr:
         res += 1
   return res


def dont_give_me_five(start, end):
   return sum('5' not in str(i) for i in range(start, end + 1))


def dont_give_me_five(start,end):
   return len([num for num in range(start, end+1) if '5' not in str(num)])


def dont_give_me_five(start,end):
   return len(filter(lambda x: "5" not in str(x), range(start, end + 1)))


print(dont_give_me_five(1,9), 8)
print(dont_give_me_five(4,17), 12)
print(dont_give_me_five(40, 60), 10)

File: PHP/7php 1/dgmf.php
<?php
/*
|--------------------------------------------------------------------------
| Don't give me five!
|--------------------------------------------------------------------------
|
*/

function dont_give_me_five($start, $end) {
   $res = 0;
   for ($i=$start; $i <= $end; $i++) { 
      $str = str_split(strval($i));
      if (in_array('5', $str) == false) {
         $res += 1;
      }
   }
   return $res;
}


function dont_give_me_five($start, $end) {
   return count(array_filter(range($start, $end), function($iNumber){
      return strpos((string)$iNumber, '5') === false;
   }));
}


function dont_give_me_five($start, $end) {
   for($i = $start;$i <= $end;$i++){
      if(strstr($i, "5") != true ){
         $k[] = $i;
      } 
   }
   return count($k);
}


function dont_give_me_five($start, $end) {
   return count(preg_grep('/[5]/',range($start, $end),PREG_GREP_INVERT));
}


function dont_give_me_five($start, $end) {
   return count(array_filter(range($start, $end), function($n){
      return !preg_match('~5~', $n); 
   }));
}


print_r(dont_give_me_five(1, 9). ' ' . 8 .'
');
print_r(dont_give_me_five(4, 17). ' ' . 12 .'
');
print_r(dont_give_me_five(40, 60). ' ' . 10 .'
');